lib/modules/apostrophe-areas/public/js/user.js

```diff
@@ -323,6 +323,41 @@ apos.define('apostrophe-areas', {
       return editors;
     };
 
+    // Loads the virtual areas inside a modal.
+    self.loadVirtualArea = function () {
+      var $virtualAreas = $('[data-virtual-area]');
+      var virtualAreaLength = $virtualAreas.length;
+      var virtualAreaLoadedCount = 0;
+      // Do nothing if there are no virtual areas
+      if ($virtualAreas.length > 0) {
+        // Disable the modal controls till every area is loaded. Clicking on Save fails without
+        // loading the virtual area.
+        apos.modalSupport.toggleModalControls(true);
```

Comment:
Should be a method of modals, which will be easy to call as self.toggleControls if this entire algorithm moves into a method of modals.